pyspark/bigdl/contrib/onnx2bigdl/_op_translations.py

```diff
@@ -0,0 +1,289 @@
+
+# coding: utf-8
+""" Module for translating ONNX operators into Mxnet operatoes"""
+# pylint: disable=unused-argument,protected-access
+import numpy as np
+from bigdl.nn.layer import *
+from translation_utils import *
+
+from onnx.numpy_helper import to_array
+
+
+# Method definitions for the callable objects mapped in the import_helper module
+def avg_pooling(inputs, prev_modules, attrs, outputs):
+
+	auto_pad = attrs.get('auto_pad', 'NOTSET')
+	ceil_mode = True if attrs.get('ceil_mode', 0) == 1 else False
+	count_include_pad = True if attrs.get('count_include_pad', 0) == 1 else False
+	kernel_shape = attrs.get('kernel_shape', (0, 0))
+	pads = attrs.get('pads', [0, 0, 0, 0])
+	strides = attrs.get('strides', [0, 0])
+
+	kernel_width, kernel_height = kernel_shape[:2]
+	stride_width, stride_height = strides[:2]
+	padding_width, padding_height = pads[:2]
+
+	_, data_tensor_shape = inputs[0]
+	input_height, input_width = data_tensor_shape[-2:]
+
+	output_height = calc_output_shape(input_height, kernel_height, 
+		padding = padding_height, stride = stride_height, ceil_mode = ceil_mode)
+	output_width = calc_output_shape(input_width, kernel_width, 
+		padding = padding_width, stride = stride_width, ceil_mode = ceil_mode)
+
+	out_tensor_shape = list(data_tensor_shape)
+	out_tensor_shape[-2] = output_height
+	out_tensor_shape[-1] = output_width
+	out_tensor_shape = tuple(out_tensor_shape)
+
+	module = SpatialAveragePooling(kernel_width, kernel_height,
+		dw = stride_width, dh = stride_height, pad_w = padding_width, pad_h = padding_height,
+		ceil_mode = ceil_mode, count_include_pad = count_include_pad)(prev_modules)
+
+	return module, [out_tensor_shape]
+
+
+def batch_norm(inputs, prev_modules, attrs, outputs):
+
+	epsilon = attrs.get('epsilon', 1e-05)
+	momentum = attrs.get('momentum', 0.9)
+
+	_, data_tensor_shape = inputs[0]
+	scale_tensor_val, _ = inputs[1]
+	bias_tensor_val, _ = inputs[2]
+
+	n_output = data_tensor_shape[1]
+	out_tensor_shape = data_tensor_shape
+
+	module = SpatialBatchNormalization(n_output, eps = epsilon)(prev_modules)
+
+	return module, [out_tensor_shape]
+
+
+def concat(inputs, prev_modules, attrs, outputs):
+
+	axis = attrs.get('axis')
+	_, data_tensor_shape = inputs[0]
+
+	dim_rank = 0
+	for i in range(len(inputs)):
+		_, curr_input_shape = inputs[i]
+		for j in range(len(data_tensor_shape)):
+			if axis != j:
+				if curr_input_shape[i] != data_tensor_shape[i]:
+					raise ValueError()
```

Comment:
Is this logic checking the shape of all inputs? Maybe need more details about the `ValueError()`.